src/owner.rs

```diff
@@ -0,0 +1,357 @@
+                                  to query an organization that owns this \
+                                  crate. You may need to re-authenticate on \
+                                  crates.io to grant permission to read \
+                                  github org memberships. Just go to \
+                                  https://crates.io/login"));
+            }
+            _ => {
+                return Err(internal(format!("didn't get a 200 result from github: {}",
+                                        resp)))
+            }
+        }
+
+        #[derive(RustcDecodable)]
+        struct Membership {
+            state: String,
+        }
+        let json = try!(str::from_utf8(resp.get_body()).ok().chain_error(||{
+            internal("github didn't send a utf8-response")
+        }));
+        let membership: Membership = try!(json::decode(json).chain_error(|| {
+            internal("github didn't send a valid json response")
+        }));
+
+        // There is also `state: pending` for which we could possibly give
+        // some feedback, but it's not obvious how that should work.
+        Ok(membership.state == "active")
+    }
+}
+
+impl Model for Team {
+    fn from_row(row: &Row) -> Self {
+        Team {
+            cargo_id: row.get("id"),
+            name: row.get("name"),
+            github_id: row.get("github_id"),
+        }
+    }
+
+    fn table_name(_: Option<Self>) -> &'static str { "teams" }
+}
+
+impl Owner {
+    /// Finds the owner by name, failing out if it doesn't exist.
+    /// May be a user's GH login, or a full team name. This is case
+    /// sensitive.
+    pub fn find_by_name(conn: &Connection, name: &str) -> CargoResult<Owner> {
+        let owner = if name.contains(":") {
+            Owner::Team(try!(Team::find_by_name(conn, name).map_err(|_|
+                human(format!("could not find team with name {}", name))
+            )))
+        } else {
+            Owner::User(try!(User::find_by_login(conn, name).map_err(|_|
+                human(format!("could not find user with login `{}`", name))
+            )))
+        };
+        Ok(owner)
+    }
+
+    /// Find the owner by name, with the intent of adding it as an owner.
+    ///
+    /// This differs from find_by_name in that in the case of a Team,
+    /// it will verify the req_user is on the team first.
+    ///
+    /// If the req_user is on the Team, it will create the team in the DB
+    /// if it is not already present. When this occurs, this will set the
+    /// One True Name of the team. All future references to the team must
+    /// use the exact casing provided here. If a different casing is provided
+    /// to this method, we may still succeed if Github returns us the same ID
+    /// as the One True Name. However in this case, the One True Name will
+    /// still be selected.
+    pub fn find_by_name_for_add(conn: &Connection, name: &str, req_user: &User)
```

Comment:
This is only called in one location, so perhaps the logic for checking whether the user is a member of a team can be inlined at that location? Either that or otherwise add a method to `Owner` querying whether a user is a member of that owner.